debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSnapshotChangeEventSource.java

```diff
@@ -0,0 +1,599 @@
+            });
+        }
+        finally {
+            taskContext.getConnectionContext().shutdown();
+        }
+
+        return new SnapshottingTask(replicaSetSnapshots);
+    }
+
+    private void delaySnapshotIfNeeded(ChangeEventSourceContext context) throws InterruptedException {
+        Duration snapshotDelay = connectorConfig.getSnapshotDelay();
+
+        if (snapshotDelay.isZero() || snapshotDelay.isNegative()) {
+            return;
+        }
+
+        Timer timer = Threads.timer(Clock.SYSTEM, snapshotDelay);
+        Metronome metronome = Metronome.parker(ConfigurationDefaults.RETURN_CONTROL_INTERVAL, Clock.SYSTEM);
+
+        while (!timer.expired()) {
+            if (!context.isRunning()) {
+                throw new InterruptedException("Interrupted while awaiting initial snapshot delay");
+            }
+
+            LOGGER.info("The connector will wait for {}s before proceeding", timer.remaining().getSeconds());
+            metronome.pause();
+        }
+    }
+
+    protected SnapshotContext prepare(ChangeEventSourceContext sourceContext) throws Exception {
+        return new MongoDbSnapshotContext();
+    }
+
+    protected void complete(SnapshotContext snapshotContext) {
+    }
+
+    private void snapshotReplicaSet(ChangeEventSourceContext sourceContext, SnapshotContext ctx, ReplicaSet replicaSet) throws InterruptedException {
+        MongoPrimary primaryClient = null;
+        try {
+            primaryClient = establishConnectionToPrimary(replicaSet);
+            if (primaryClient != null) {
+                createDataEvents(sourceContext, ctx, replicaSet, primaryClient);
+            }
+        }
+        finally {
+            if (primaryClient != null) {
+                primaryClient.stop();
+            }
+        }
+    }
+
+    private MongoPrimary establishConnectionToPrimary(ReplicaSet replicaSet) {
+        return connectionContext.primaryFor(replicaSet, taskContext.filters(), (desc, error) -> {
+            // propagate authorization failures
+            if (error.getMessage() != null && error.getMessage().startsWith(AUTHORIZATION_FAILURE_MESSAGE)) {
+                throw new ConnectException("Error while attempting to " + desc, error);
+            }
+            else {
+                LOGGER.error("Error while attempting to {}: ", desc, error.getMessage(), error);
+                throw new ConnectException("Error while attempting to " + desc, error);
+            }
+        });
+    }
+
+    private boolean isInitialSyncExpected(MongoPrimary primaryClient, ReplicaSetOffsetContext offsetContext) {
+        boolean performSnapshot = true;
+        if (offsetContext.hasOffset()) {
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info("Found existing offset for replica set '{}' at {}", offsetContext.getReplicaSetName(), offsetContext.getOffset());
+            }
+            performSnapshot = false;
+            if (connectionContext.performSnapshotEvenIfNotNeeded()) {
```

Comment:
Hardcoded return false value - etiher a bug or code not needed

Reply:
I guess that depends on how you look at it :P.

MongoDB does not have a concept of `SnapshotMode.ALWAYS` like other connectors, it only supports _never_ or _initial_, so I'd say this code that isn't needed.  wdyt?

Reply:
For now I removed this dead code in commit 0158b56.